frontend/src/components/shared/TableOfContents.vue

```diff
@@ -42,26 +44,30 @@ export default {
   },
   methods: {
     hasActiveSubsection({ subsections }) {
-      return (subsections ?? []).map(s => s.link).includes(this.$route.hash);
+      const links = (subsections ?? []).map(s => s.link);
+      return links.includes(this.$route.hash) || links.some(l => l.includes(this.$route.path));
+    },
+    isActiveLink(link) {
+      return link === this.$route.path || this.$route.hash.replace('#', '') === link.split('#')[1];
     },
   },
 };
 </script>
 
 <style lang="scss" scoped>
```

Comment:
Since we have learnt that there is no performance penalties of using `scoped`. I would actually advocate for adding them to all components unless they specifically include styles that should be global. In fact earlier in the day we [added `scoped` to one component](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1126/commits/990b20d99b8836ecba28112f22566471251f1194) because otherwise the Bulma classes override the styles for `fileNameBox` in: `class="fileNameBox tags has-addons is-centered mb-0"`.

I'm happy to discuss more about this though but it would be nice if we can avoid it as well as this potential change for now.

Reply:
I see your point. I was looking at the first style below, and there didn't seem to be a need for `scoped` since it's mapped via `id`. Let's keep this as is!